Homework/2/Program.cs

```diff
@@ -0,0 +1,470 @@
+            if (endOfArray)
+            {
+                return userArray.Length;
+            }
+            else if (startOfArray)
+            {
+                return 0;
+            }
+            else if (!endOfArray && !startOfArray)
+            {
+                string indexStr = PromptInput("Enter index:");
+                int.TryParse(indexStr, out int index);
+
+                if (index < 0 || index >= userArray.Length)
+                {
+                    return -1;
+                }
+                else {
+                    return index;
+                }
+            }
+            return -1;
+        }
+
+
+        //Remove user from userArray at index.
+        static void RemoveUser(int index) {
+            string[][] newArray = new string[userArray.Length - 1][];
+
+            for (int i = 0; i < newArray.Length; i++)
+            {
+                if (i <index)
+                {
+                    newArray[i] = userArray[i];
+                }else if (i == index)
+                {
+                    newArray[i] = userArray[i + 1];
+                }else if (i > index)
+                {
+                    newArray[i] = userArray[i + 1];
+                }
+            }
+
+            userArray = newArray;
+        }
+
+        //Sorting users
+        static void SortUsers() {
+            Console.Clear();
+
+            bool ascendingOrder = ReturnSortingOrder();
+
+            if (ascendingOrder)
+            {
+                SortUserArrayInAscOrder();
+            }
+            else {
+                SortUserArrayInDesOrder();
+            }
+
+            WriteUsersToFile();
+            ListUsers();
+        }
+
+        // Function to swap values in userArray
+        static void UserArrayValueSwap(int x, int y) {
+            string[] temp;
+            temp = userArray[x];
+            userArray[x] = userArray[y];
+            userArray[y] = temp;
+        }
+
+        //Ascending order sorting function
+        static void SortUserArrayInAscOrder() {
+            for (int j = 0; j < userArray.Length; j++)
+            {
+                for (int i = 0; i < userArray.Length - 1; i++)
+                {
+                    if (userArray[i][0].CompareTo(userArray[i + 1][0]) > 0)
+                    {
+                        UserArrayValueSwap(i, i + 1);
+                    }
+                }
+            }
+        }
+
+        //Descending order sorting function
+        static void SortUserArrayInDesOrder() {
+            for (int j = 0; j < userArray.Length; j++)
+            {
+                for (int i = 0; i < userArray.Length - 1; i++)
+                {
+                    if (userArray[i][0].CompareTo(userArray[i + 1][0]) < 0)
+                    {
+                        UserArrayValueSwap(i, i + 1);
+                    }
+                }
+            }
+        }
+
+        //True for ascending/ false for descending
+        static bool ReturnSortingOrder()
```

Comment:
We haven't learned about it yet but it might be nice to return an enum here to keep the code more readable.  For example, is "false" descending or ascending?  Another way to tackle it might be to rename this method to something like `GetShouldSortAscending`